docs/source/api/insights.rst

```diff
@@ -0,0 +1,767 @@
+
+1. `ESP_OK` : On success
+2. Error in case of failure
+
+Variables.addMAC
+****************
+
+Add the MAC address variable
+
+.. code-block:: arduino
+
+    esp_err_t addMAC(const char *key, uint8_t *mac);
+
+* ``key`` : Key of the variable
+* ``mac`` : Array of length 6 i.e 6 octets of mac address
+
+This function will return
+
+1. `ESP_OK` : On success
+2. Error in case of failure
+
+Variables.addString
+*******************
+
+Add the variable of data type string
+
+.. code-block:: arduino
+
+    esp_err_t addString(const char *key, const char *str);
+
+* ``key`` : Key of the variable
+* ``str`` : Value of the variable
+
+This function will return
+
+1. `ESP_OK` : On success
+2. Error in case of failure
+
+Variables.initNetworkVariables
+******************************
+
+Initialize the network variables
+Below listed Wi-Fi and IP parameters are collected and reported to cloud on change.
+Wi-Fi connection status, BSSID, SSID, channel, authentication mode,
+Wi-Fi disconnection reason, IP address, netmask, and gateway.
+
+.. code-block:: arduino
+
+    esp_err_t initNetworkVariables();
+
+This function will return
+
+1. `ESP_OK` : On success
+2. Error in case of failure
+
+Variables.deinitNetworkVariables
+********************************
+
+Deinitialize the network variables
+
+.. code-block:: arduino
+
+    esp_err_t deinitNetworkVariables();
+
+This function will return
+
+1. `ESP_OK` : On success
+2. Error in case of failure
+
+Variables.deinit
+****************
+
+Deinitialize the diagnostics variables
+
+.. code-block:: arduino
+
+    esp_err_t deinit();
+
+This function will return
+
+1. `ESP_OK` : On success
+2. Error in case of failure
+
+
+ESP Insights Diagnostics API
+----------------------------
+
+`Diagnostics` object of `DiagnosticsClass` class exposes API's for reporting logs to the Insights Cloud
+This class is defined in "Diagnostics.h" header file.
+
+
+Diagnostics.initLogHook
+***********************
+
+Initialize diagnostics log hook
+
+.. code-block:: arduino
+
+    esp_err_t initLogHook(esp_diag_log_write_cb_t write_cb, void *cb_arg = NULL);
+
+* ``write_cb`` : Callback function to write diagnostics data
+
+This function will return
+
+1. `ESP_OK` : On success
+2. Error in case of failure
+
+Diagnostics.enableLogHook
+*************************
+
+Enable the diagnostics log hook for provided log type
+
+.. code-block:: arduino
+
+    void enableLogHook(uint32_t type);
+
+* ``type`` : Log type to enable, can be the bitwise OR of types from esp_diag_log_type_t
+
+This function will return
+
+1. `ESP_OK` : On success
+2. Error in case of failure
+
+Diagnostics.disableLogHook
+**************************
+
+Disable the diagnostics log hook for provided log type
+
+.. code-block:: arduino
+
+    void disableLogHook(uint32_t type);
+
+* ``type`` : Log type to disable, can be the bitwise OR of types from esp_diag_log_type_t
+
+This function will return
+
+1. `ESP_OK` : On success
+2. Error in case of failure
```

Comment:
Add the example at the end by adding a literal include to the example.

I.e:

```
.. literalinclude:: ../../../libraries/ESP32/examples/HallSensor/HallSensor.ino
    :language: arduino
```

Reply:
PTAL

Reply:
Thanks!